### ISSUE TITLE
Multi metric evaluation and hyperparameters selection

### ISSUE THREAD

**Issue by aminevsaziz** [hyperopt/hyperopt#435]
I'm stumbling a dead wall on how to optimize a scikit model/models based on multiple metric(accuracy, logloss and kappa) using hyperopt. I've be looking for an example or something that may help so I end up being here to ask if someone know how or even knows a trick to make this possible.
I appreciate any help or simple demonstration.

**Comment by brandonschabell** [comment]
What I have done, which seems to work well, is to use a hyperopt loss function that optimizes a metric (for me, I've been looking at f2 scores, but you could look at accuracy instead if that is your main concern.) You could then use a logloss for your model itself.

A crude example:

```python
def objective(x):
  model = sklearn.some_model(x).fit(data, loss='logloss')
  accuracy = model['acc']
  return {
    'loss': 1 - accuracy,
    'status': STATUS_OK,

trials = Trials()
best = fmin(objective, space=hp.uniform('x', 0, 100), algo=tpe.suggest, max_evals=100, trials=trials)
```

**Comment by brandonschabell** [comment]
Alternatively, you could create an objective function with a composite loss.

```python
def objective(x):
  model = sklearn.some_model(x).fit(data, loss='logloss')
  accuracy = model['acc']
  precision = model['pre']
  recall = model['rec']
  loss = (1 - accuracy) * (1 - precision) * (1 - recall)
  return {
    'loss': loss ,
    'status': STATUS_OK
  }
```

**Comment by recursionbane** [comment]
I've found that specifying weights and constructing a composite loss function works well.

Something like this at the tail end of the objective function:

```
# Use supplied parameters to compute current values of each metric
...
# Then, compute compound loss
composite_loss = 0
for metric in metrics:
  single_loss = metrics[metric]['ideal'] - metrics[metric]['current']
  composite_loss = single_loss * metrics[metric]['weight']
return composite_loss
```

This allows you to specify a weight of zero for "don't-care" metrics, and arbitrary (and even overlapping) weights for other metrics.
